index.html

```diff
@@ -3409,6 +3409,45 @@ <h3>Syntactic Sugar</h3>
         </section>
 
       </section>
+      <section>
+        <h3>Media Types</h3>
+
+        <p>
+          Media types as defined in [[RFC6838]] serve a useful purpose with verifiable 
+          credentials, especially regarding identification of a particular syntax 
+          that is in use with a verifiable credential.
+        </p>
+        <p>
+          Syntaxes SHOULD be identified by a media type, and certain conventions as 
+          outlined in this section should be followed when defining or using media types 
+          with verifiable credentials.
+        </p>
+        <p>
+          Any media types associated with the core data model are listed in the section on
+          <a href="#iana-considerations">IANA Considerations</a> for registration with IANA.
+        </p>
+        <p>
+          When defining a media type for use with verifiable credentials — for instance, in a 
+          specification that defines a specific syntax — use of the term `verifiable` implies
+          the presence of a <a href="#proofs-signatures">proof</a> with the credential.  
```

Comment:
```suggestion
          the presence of a <a href="#proofs-signatures">proof</a> with the credential either embedded or external to the credential.  
```

@OR13 note that i linked directly to the section that calls out both embedded AND external proofs intentionally here.  perhaps some additional language as in this suggestion would help

Reply:
> At least one proof mechanism, and the details necessary to evaluate that proof, MUST be expressed for a credential or presentation to be a verifiable credential or verifiable presentation; that is, to be verifiable.

I feel like this spec text is being disputed by @iherman @TallTed @dlongley @msporny and possibly others.

> Media types MAY be registered with more than one suffix appended to
> the base subtype name.  The suffixes MUST be interpreted as ordered.
> Valid media type names containing a structured suffix are built from
> right to left (not left to right).  Characters on the left-most side
> of the left-most "+" in a subtype name specify the base subtype name.
> Characters to the right of each "+" in a subtype name denote
> additional structured syntax suffixes.

> Media types with more than one suffix MUST be registered according to
> the procedure defined in [RFC6838].  A new base subtype name MUST
> only be registered with suffix combinations that are already
> registered in their own right in the Structured Syntax Suffixes
> registry (https://www.iana.org/assignments/media-type-structured-suffix/media-type-structured-suffix.xhtml).

- https://datatracker.ietf.org/doc/draft-ietf-mediaman-suffixes/

> While it is possible for a given media type to be assigned additional
> names, the use of different names to identify the same media type is
> discouraged.

- https://www.rfc-editor.org/rfc/rfc6838#section-4.2

(🔥 let us not register 2 media types that can be used for the "same bytes").

> Media types that make use of a named structured syntax SHOULD use the
> appropriate registered "+suffix" for that structured syntax when they
> are registered.  By the same token, media types MUST NOT be given
> names incorporating suffixes for structured syntaxes they do not
> actually employ. "+suffix" constructs for as-yet unregistered
> structured syntaxes SHOULD NOT be used, given the possibility of
> conflicts with future suffix definitions.

- https://www.rfc-editor.org/rfc/rfc6838#section-4.2.8

(🔥 let us make sure things are actually verifiable or credentials).

Based on this comment: https://github.com/w3c/vc-data-model/issues/1044#issuecomment-1446805038

> `VerifiableCredential` might contain `proof` (modified since `proof` is actually optional in `VerifiableCredential`.)
>  `VerifiableCredential` is a subtype of `Credential`

processing from right to left...

is the `credential` ... `json` ?
is the `credential ... `ld+json`?
is the `credential ... `verifiable+ld+json`? (does this violate the MUST NOT or SHOULD NOT from the RFC?)

Compared to...

is the `verifiable` ... `json` ?
is the `verifiable` ... `ld+json` ?
is the `verifiable` ... `credential+ld+json` ? (does this violate the MUST NOT or SHOULD NOT from the RFC?)

Reply:
>>At least one proof mechanism, and the details necessary to evaluate that proof, MUST be expressed for a credential or presentation to be a verifiable credential or verifiable presentation; that is, to be verifiable.
>
> [@OR13] [I feel like this spec text is being disputed by @iherman @TallTed @dlongley @msporny and possibly others.](https://github.com/w3c/vc-data-model/pull/1055#discussion_r1122357911)

What a strange feeling... as I am not disputing that sentence at all.

I cannot make heads or tails of the remainder of your comment. Maybe try writing full sentences, including what brought you to quote the chunks of draft and ratified RFCs?

A reminder: "Classes" and "Types" do not _require_ Media Types to be defined and useful. A "tiger" is a subtype of "cat", which is a subtype of "mammal", etc. None of these have Media Types, but they are all useful descriptors. A "CompactDisc" is disjoint from "CassetteTape" and both are disjoint from "VinylLP", but all are subtypes of "Audio Recording Media", so all might be considered Media Types ... but this typing is not handled by RFC nor IANA.

Reply:
@TallTed The line of argument you have here: https://github.com/w3c/vc-data-model/pull/1055#discussion_r1123655680

Regarding `proof`, is what leads me to believe that you do not see a difference in information between `credential` and `verifiable credential`.

Reply:
@OR13,

> let us not register 2 media types that can be used for the "same bytes"

This is literally what the `+` syntax does in media types. You can describe the string `{}` using either `application/json` OR `application/ld+json` today. There are numerous other examples with `+xml` suffixes and others. This subclassing is done on purpose to enable many different use cases.

@TallTed has done an excellent job in various threads explaining these rules -- but I fear there continue to be misunderstandings in various places.

Reply:
I believe that @dlongley @TallTed @iherman and @msporny are asserting that assuming we register `credential+ld+json`, `verifiable+credential+ld+json` or `verifiable-credential+ld+json`, they would all / both be valid content types for the following:

```json
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://w3id.org/security/suites/bls12381-2020/v1"
  ],
  "id": "http://example.edu/credentials/3732",
  "type": ["VerifiableCredential"],
  "issuer": "https://example.edu/issuers/14",
  "issuanceDate": "2010-01-01T19:23:24Z",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21"
  },
  "proof": {
    "type": "BbsBlsSignature2020",
    "created": "2010-01-01T19:23:24Z",
    "verificationMethod": "did:example:123#key-0",
    "proofPurpose": "assertionMethod",
    "proofValue": "uBncn..."
  }
}
```

My point is that we don't need to register 3 media types for the same bytes, we only need to register `verifiable+credential+ld+json` or `verifiable-credential+ld+json` .

Then the whole "is proof allowed in `credential+ld+json` question" goes away, since `credential+ld+json` is no longer a valid media type.

The alternative proposal is `proof` is not allowed in `credential+ld+json`  and is optional in `verifiable+credential+ld+json` and we must register both, this was my original proposal, let us call it the "2 media type proposal"

@msporny indicated he was in favor of just registering `verifiable+credential+ld+json` and not registering `credential+ld+json`, let us call this the "one media type proposal".

Here are some proposals to run in the next working group call:

PROPOSAL: The 1 media type for vc proposal, the working group will register `verifiable+credential+ld+json` and the working group will not register `credential+ld+json`.

PROPOSAL: The 2 media type for vc proposal, the working group will register `verifiable+credential+ld+json` and the working group will register `credential+ld+json`.

Reply:
> I fear there continue to be some misunderstandings.

I share your fear. In spades.

Reply:
Given the comment from @TallTed https://github.com/w3c/vc-data-model/pull/1055#discussion_r1123655680 , then I would suggest we amend the proposals to be the following: 

PROPOSAL: The 1 media type for vc proposal, the working group will register `verifiable-credential+ld+json` and the working group will not register `credential+ld+json`.

PROPOSAL: The 2 media type for vc proposal, the working group will register `verifiable+credential+ld+json` and the working group will register `credential+ld+json`.

Reply:
I would be a +1 for the first proposal directly above this comment (initially proposed by @OR13 and then refined outlined by @andresuribe87), with the caveat that the name of the media type is `vc+ld+json` (where the bikeshedding thereof doesn't need to hold up the decision).

Reply:
@msporny I'm not arguing you can't have different media types for the same bytes.... I am arguing against registered multiple media types for the same bytes with indistinguishable definitions in our spec.

+1 to `vc+ld+json`

Reply:
> I am arguing against registered multiple media types for the same bytes with indistinguishable definitions in our spec.

The first question should be whether there's value in having the second media type, whether or not it is a supertype of the type we're primarily focused on.  If there is such value (as I believe there is), then if it is a supertype (as I believe it is), I think we should register both, _or only the supertype_.  (If it were a subtype, we might not register it, but there would be good reason to speak of it and to suggest a media type for it.)

Remember that media types are not actively restrictive nor actively prescriptive in most applications, but are primarily *hints* about how to process the bytes within the resource labeled with that media type.  In the world of the web, they're helpful for various tools to make known to a web server what form of resource those tools prefer such that the server can provide that, if available.  (See [RFC2295](https://www.rfc-editor.org/rfc/rfc2295), among others.)  It's perfectly valid for my super app to process resources of *every* media type in exactly the same way, if users of my super app find value in what my app outputs based on those inputs.

Reply:
@TallTed does this 3rd proposal capture your alternate single registration proposal? :

PROPOSAL:The 1 media type for `credential` proposal, the working group will register `credential+ld+json` and the working group will not register `verifiable+credential+ld+json`.